src/domains/shared-app/api/session-wallet/v1/_base_policy.xml.tpl

```diff
@@ -1,225 +1,168 @@
-                  <value>application/json</value>
+                    <value>application/json</value>
                 </set-header>
                 <set-body>
                     {
                         "title": "Error starting session",
                         "status": 502,
-                        "detail": "Cannot tokenize user fiscal code: PM start-session fiscalCode is null"
+                        "detail": "There was an error while getting user info"
                     }
                 </set-body>
-              </return-response>
-          </when>
-          <otherwise>
-              <!-- Post Token PDV : START-->
-              <send-request ignore-error="true" timeout="10" response-variable-name="pdv-token" mode="new">
-                <set-url>${pdv_api_base_path}/tokens</set-url>
-                <set-method>PUT</set-method>
-                <set-header name="x-api-key" exists-action="override">
-                    <value>{{wallet-session-personal-data-vault-api-key}}</value>
+            </return-response>
+        </when>
+      </choose>
+      <set-variable name="userAuthBody" value="@(((IResponse)context.Variables["user-auth-body"]).Body.As<JObject>())" />
+      <set-variable name="userFiscalCode" value="@{
+        String userFiscalCode = ((JObject)context.Variables["userAuthBody"])?["fiscal_code"]?.ToString();
+        return userFiscalCode;
+    }" 
+    />
+      <!-- Get User IO : END-->
+
+      <!-- user fiscal code tokenization with PDV START -->
+      <!-- Post Token PDV : START-->
+      <send-request ignore-error="true" timeout="10" response-variable-name="pdv-token" mode="new">
+        <set-url>${pdv_api_base_path}/tokens</set-url>
+        <set-method>PUT</set-method>
+        <set-header name="x-api-key" exists-action="override">
+            <value>{{wallet-session-personal-data-vault-api-key}}</value>
+        </set-header>
+        <set-body>@{
+          return new JObject(
+                  new JProperty("pii", (string)context.Variables["userFiscalCode"])
+              ).ToString();
+            }</set-body>
+      </send-request>
+      <choose>
+        <when condition="@(((IResponse)context.Variables["pdv-token"]).StatusCode != 200)">
+          <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+              <set-body>
+              {
+                  "title": "Error starting session",
+                  "status": 502,
+                  "detail": "Error during fiscal code tokenization"
+              }
+          </set-body>
+          </return-response>
+        </when>
+      </choose>
+
+      <set-variable name="pdvToken" value="@(((IResponse)context.Variables["pdv-token"]).Body.As<JObject>())" />
+      <!-- used as jwt claims  https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims -->      
+      <set-variable name="userId" value="@((string)((JObject)context.Variables["pdvToken"])["token"])" />
```

Comment:
```suggestion
      <!-- used as jwt claims  https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims -->  
      <set-variable name="userId" value="@(((IResponse)context.Variables["pdv-token"]).Body.As<JObject>()["token"])" />
```
Since here we are reading only one variable from response body we can reduce one variable creation overhead assigning user id in a single step.
What do you think?

Reply:
it's a good point